lib/blob/blobstore.c

```diff
@@ -2777,6 +2777,42 @@ blob_calculate_lba_and_lba_count(struct spdk_blob *blob, uint64_t io_unit, uint6
 	}
 }
 
+static inline bool
+blob_ancestor_calc_lba_and_lba_count(struct spdk_blob *blob, uint64_t io_unit, uint64_t length,
+				     uint64_t *lba, uint64_t *lba_count)
+{
+	struct spdk_blob *orig_blob = blob;
+	*lba_count = length;
+	if (!bs_io_unit_is_allocated(orig_blob, io_unit)) {
```

Comment:
We don't need to check this on original blob again, and directly start traversing parent blobs provided we are calling this function if is_allocated is `false` from existing checks.

Reply:
`is_allocated` check is done after `blob_ancestor_calc_lba_and_lba_count` is called and `is_allocated` is replaced with newly added function. If new function only checks for parent blob, then it will return false if child has not snapshot.

Reply:
Done the changes. PTAL